control-plane/plugin/src/resources/node.rs

```diff
@@ -118,23 +118,58 @@ fn drain_labels_from_state(ds: &CordonDrainState) -> Vec<String> {
 impl Cordoning for Node {
     type ID = NodeId;
     async fn cordon(id: &Self::ID, label: &str, output: &OutputFormat) {
-        match RestClient::client()
-            .nodes_api()
-            .put_node_cordon(id, label)
-            .await
-        {
-            Ok(node) => match output {
-                OutputFormat::Yaml | OutputFormat::Json => {
-                    // Print json or yaml based on output format.
-                    utils::print_table(output, node.into_body());
+        // is node already cordoned with the label?
+        let already_has_cordon_label: bool =
+            match RestClient::client().nodes_api().get_node(id).await {
+                Ok(node) => {
+                    let node_body = &node.into_body();
+
+                    match &node_body.spec.as_ref().unwrap().cordondrainstate {
```

Comment:
handle `unwrap`, if no spec then error with node is not registered.

Reply:
fixed